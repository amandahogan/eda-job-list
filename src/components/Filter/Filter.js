import React, { PropTypes } from 'react'
import { Button, Dropdown, Icon, Label, Menu } from 'semantic-ui-react'

class Filter extends React.Component {
  constructor (props) {
    super(props)
    this.state = {selectedCategoryId: null, selectedTagValue: null}
    this.onCategoryChange = this.onCategoryChange.bind(this)
    this.onTagChange = this.onTagChange.bind(this)
  }

  onCategoryChange (event, data) {
    this.setState({
      selectedCategoryId: data.value,
      selectedTagValue: null
    })
  }

  onTagChange (event, data) {
    this.setState({
      selectedTagValue: data.value
    })
  }

  render () {
    const { categories, tags } = this.props
    const categoryOptions = categories.map(function (category) {
      return {
        key: category.id,
        text: category.name,
        value: category.id
      }
    })
    const tagOptions = tags
      .filter(tag => {
        return tag.category.id === this.state.selectedCategoryId
      })
      .map(tag => {
        return {
          text: tag.value,
          value: tag.value
        }
      })

    return (
      <div className='Filter'>
        <Menu fluid>
          <Dropdown
            fluid
            options={categoryOptions}
            onChange={this.onCategoryChange}
            placeholder='Category'
            selection
          />
          <Dropdown
            allowAdditions
            fluid
            options={tagOptions}
            onChange={this.onTagChange}
            placeholder='Tag'
            search
            selection
          />
          <Button
            basic
            circular
            disabled={this.state.selectedCategoryId === null || this.state.selectedTagValue === null}
            icon='plus'
            onClick={() => {
              this.props.addIncludeTag(this.state.selectedCategoryId, this.state.selectedTagValue)
            }}
          />
          <Button
            basic
            circular
            disabled={this.state.selectedCategoryId === null || this.state.selectedTagValue === null}
            icon='minus'
            onClick={() => {
              this.props.addExcludeTag(this.state.selectedCategoryId, this.state.selectedTagValue)
            }}
          />
        </Menu>
        <Label as='a'>
          <Icon color='green' name='check' />
          Javascript
          <Icon name='delete' />
        </Label>
        <Label as='a'>
          <Icon color='red' name='ban' />
          AngularJS
          <Icon name='delete' />
        </Label>
      </div>
    )
  }
}

export default Filter

Filter.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired
    })
  ).isRequired
}
