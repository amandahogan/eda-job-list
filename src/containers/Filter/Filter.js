import { connect } from 'react-redux'

import { addExcludeTag, addIncludeTag, removeExcludeTag, removeIncludeTag } from '../../actions/columns'
import Filter from '../../components/Filter'

export default connect(
  state => {
    return {}
  },
  (dispatch, ownProps) => {
    const columnId = ownProps.column.id
    return {
      addExcludeTag: (categoryId, value) => dispatch(addExcludeTag(columnId, categoryId, value)),
      addIncludeTag: (categoryId, value) => dispatch(addIncludeTag(columnId, categoryId, value)),
      removeExcludeTag: tagId => dispatch(removeExcludeTag(columnId, tagId)),
      removeIncludeTag: tagId => dispatch(removeIncludeTag(columnId, tagId))
    }
  }
)(Filter)
