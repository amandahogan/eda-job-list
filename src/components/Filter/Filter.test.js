import { shallow } from 'enzyme'
import React from 'react'
import ReactDOM from 'react-dom'

import Filter from './Filter'

describe('<Filter />', () => {
  it('renders without crashing', () => {
    const addExcludeTag = () => {}
    const addIncludeTag = () => {}
    const removeExcludeTag = () => {}
    const removeIncludeTag = () => {}
    const div = document.createElement('div')
    const categories = [{id: 0, name: 'test'}]
    const column = {id: 0}
    const tags = []
    ReactDOM.render(<Filter categories={categories} column={column} exclude={[]} include={[]} tags={tags} addExcludeTag={addExcludeTag} addIncludeTag={addIncludeTag} removeExcludeTag={removeExcludeTag} removeIncludeTag={removeIncludeTag} />, div)
  })

  it('matches snapshot', () => {
    const addExcludeTag = () => {}
    const addIncludeTag = () => {}
    const removeExcludeTag = () => {}
    const removeIncludeTag = () => {}
    const categories = [{id: 0, name: 'test'}]
    const column = {id: 0}
    const tags = []
    expect(
      shallow(
        <Filter categories={categories} column={column} exclude={[]} include={[]} tags={tags} addExcludeTag={addExcludeTag} addIncludeTag={addIncludeTag} removeExcludeTag={removeExcludeTag} removeIncludeTag={removeIncludeTag} />).getNode()
    ).toMatchSnapshot()
  })
})
