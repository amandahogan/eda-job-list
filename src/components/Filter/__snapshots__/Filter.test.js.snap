exports[`<Filter /> matches snapshot 1`] = `
<div
  className="Filter">
  <Menu
    fluid={true}>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      fluid={true}
      icon="dropdown"
      noResultsMessage="No results found."
      openOnFocus={true}
      options={
        Array [
          Object {
            "key": 0,
            "text": "test",
            "value": "test",
          },
        ]
      }
      placeholder="Category"
      renderLabel={[Function]}
      selectOnBlur={true}
      selection={true} />
    <Input
      placeholder="Search..."
      type="text" />
    <Button
      as="button"
      basic={true}
      circular={true}
      icon="plus" />
    <Button
      as="button"
      basic={true}
      circular={true}
      icon="minus" />
  </Menu>
  <Label
    as="a">
    <Icon
      as="i"
      color="green"
      name="check" />
    Javascript
    <Icon
      as="i"
      name="delete" />
  </Label>
  <Label
    as="a">
    <Icon
      as="i"
      color="red"
      name="ban" />
    AngularJS
    <Icon
      as="i"
      name="delete" />
  </Label>
</div>
`;
